src/controllers/carpool.ctrl.ts

```diff
@@ -0,0 +1,93 @@
+import restify = require('restify')
+import carpoolService = require('../services/carpool.svc')
+import userCtrl = require('./create-user.ctrl');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+
+module CarpoolController {
+  export interface OutputJSON {
+    name: string;
+    description: string;
+    owner: userCtrl.OutputJSON;
+    campus: any; // This should be of type CampusController.OutputJSON
+    participants: Array<userCtrl.OutputJSON>;
+    href: string;
+  }
+
+  export function toOutputJSON(carpool:models.Carpool):OutputJSON {
+    return {
+      'name': carpool.name,
+      'description': carpool.description,
+      'owner': userCtrl.toOutputJSON(carpool.owner),
+      'campus': carpool.campus, // TODO: shoudl be campusCtrl.toOutputJSON
+      'participants': carpool.participants.map(userCtrl.toOutputJSON),
+      'href': '/carpools/' + carpool.id
+    };
+  }
+  export function createCarpool(
+    req:restify.Request, res:restify.Response, next:restify.Next) {
+      var carpoolName:string = req.body.name;
+      var campusName:string = req.body.campus;
+      var description:string = req.body.description;
+      var owner:string = req.body.owner;
+
+      carpoolService.createCarpool(
+        carpoolName, campusName, description, owner)
+        .then((carpool) => {
+          res.send(201, toOutputJSON(carpool));
+        })
+        .catch(errors.CarpoolOwnerNotFoundException,
+          errors.CampusNotFoundException, (err) => {
+          next(new restify.NotAcceptableError(err.message));
+        })
+        .catch(errors.CarpoolExistException, (err) => {
+          next(new restify.ConflictError(err.message));
```

Comment:
If you pass an `Error` object to `next()`, restify will automatically format that into a HTTP 500 by default. The `Error` object will get converted into a JSON object. If you use the built-in restify errors, restify will set the appropriate HTTP status code. 

So if I call `next(new restify.ConflictError("foo"))`, the HTTP status would be 409.